### PR TITLE
update to only read the condensed catalog once

### LINKED ISSUE #167: Opening DSS files/catalogs in DSS perspective is very slow
This was originally reported by @hxiedwr in #163.

Opening DSS files/catalogs in DSS perspective is very slow.

@hxiedwr added:

> For "(3) Opening dss files/catalogs in DSS perspective is very slow", we should follow the example of opening svar dss file in WRIMS-Core. The speed there is normal.

@hxiedwr also shared [InstructionErrorCreated.docx](https://github.com/user-attachments/files/17778866/InstructionErrorCreated.docx)

### PR DESCRIPTION
**Pull request by adamkorynta**
Fixes #167

The condensed catalogs were getting read in DSSFileView, then read again in the DSSCatalogView. Rewrote the logic to move the progress indicators into DSSFileView and then only access the cached condensed catalog in DSSCatalogView.

Moving the progress indicators also improves the UX and moves file load off of the UI thread.

### CODE REVIEW

**Review by rma-tom — CHANGES_REQUESTED** [review]
I built a jar with the revised code, and the DTS tab in the DSS perspective didn't open properly when I tried running with it, but it works with the jar from Feature/devOps

@hxiedwr should have a look

Here's the top of the stack trace:
!ENTRY org.eclipse.e4.ui.workbench 4 0 2024-12-05 11:38:18.648
!MESSAGE
!STACK 0
java.lang.IllegalArgumentException: Catalog is empty ?
        at vista.db.dss.DSSCatalogReader.readCatalog(DSSCatalogReader.java:95)
        at vista.db.dss.DSSCatalogReader.<init>(DSSCatalogReader.java:82)
        at vista.db.dss.DSSUtil.createCatalogReader(DSSUtil.java:562)
        at vista.db.dss.DSSGroup.getInitializedGroup(DSSGroup.java:108)
        at vista.set.GroupProxy.initializeGroup(GroupProxy.java:205)
        at vista.set.GroupProxy.getNumberOfDataReferences(GroupProxy.java:77)
        at calsim.app.Project.setDVHashtable(Project.java:115)
        at gov.ca.dwr.hecdssvue.components.DataOps.setProject(DataOps.java:524)
        at gov.ca.dwr.hecdssvue.views.DTSView.createPartControl(DTSView.java:49)
        at org.eclipse.ui.internal.e4.compatibility.CompatibilityPart.createPartControl(CompatibilityPart.java:141)

**Comment by hxiedwr** [comment]
Seems vista is used here. Can RMA go ahead removed the usage of vista? I thought you already removed the dependence on vista.jar, right?

**Comment by hxiedwr** [comment]
By the way, I am not a better expert on HecLib 7 than  RMA. So you just need report the issue & fix it. If you still have issues, I am able to work on it but I am not sure if I can do it.

**Comment by adamkorynta** [comment]
@hxiedwr - the wrims engine and comparison report tool no longer have Vista dependencies, however the CalSim dependency uses Vista (which is what is getting hit in this case). We'd either need to fully remove CalSim as well, or update CalSim to no longer use Vista.

File: dwr-hecdssvue/src/main/java/gov/ca/dwr/hecdssvue/views/DSSFileView.java [hunk 1]
```diff
@@ -511,59 +567,13 @@ public void saveProject(String fn){
 	
 	public void selectFiles(){
 		selectFileNames();
-		//DataOps.setProject();
```
Comment (adamkorynta) [inline dwr-hecdssvue/src/main/java/gov/ca/dwr/hecdssvue/views/DSSFileView.java:514]: @hxiedwr - this commented out line is similar to what we are hitting. Why was this line commented out and do we even need to call it in the DTSView.createPartControl anymore?

Comment (adamkorynta) [reply dwr-hecdssvue/src/main/java/gov/ca/dwr/hecdssvue/views/DSSFileView.java:514]: https://github.com/CentralValleyModeling/wrims/commit/428f5098dc81ae91a3ed153444120ead3d65a012

**Comment by adamkorynta** [comment]
@hxiedwr - this PR did not effect the bug and we figured out that it manifests when the DSS perspective isn't already loaded. We can address the issue in a separate code change and issue.

File: dwr-hecdssvue/src/main/java/gov/ca/dwr/hecdssvue/views/DSSFileView.java (hunk 1 above)
Comment (adamkorynta) [reply dwr-hecdssvue/src/main/java/gov/ca/dwr/hecdssvue/views/DSSFileView.java:514]: I created issue: https://github.com/CentralValleyModeling/wrims/issues/178

**Review by rma-tom — APPROVED** [review]
DTS tab issue was not related to this PR. No problems identified in the code changes for this PR.

**Pull request merged**